From a92c14f17f35b0bdd0bfde7e23f7d05e5aa2c147 Mon Sep 17 00:00:00 2001
From: "Rywacki, Kamil" <kamil.rywacki@intel.com>
Date: Thu, 3 Dec 2020 15:51:46 +0100
Subject: [PATCH] PONRTSYS-8371: PON QOS - Packets not forwarded to CPU

Change how bMAC_DstEnable, bMAC_SrcEnable and nTrafficClassAlternate are set in PON QOS
---
 drivers/net/datapath/pon_qos/pon_qos_tc_trap.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/drivers/net/datapath/pon_qos/pon_qos_tc_trap.c b/drivers/net/datapath/pon_qos/pon_qos_tc_trap.c
index 72cb4000a479..e5960cff1848 100644
--- a/drivers/net/datapath/pon_qos/pon_qos_tc_trap.c
+++ b/drivers/net/datapath/pon_qos/pon_qos_tc_trap.c
@@ -229,8 +229,10 @@ static void pon_qos_trap_parse_key_eth(struct net_device *dev,
 						  FLOW_DISSECTOR_KEY_ETH_ADDRS,
 						  f->mask);
 
-		pce_rule->pattern.bMAC_DstEnable = is_zero_ether_addr(key->dst);
-		pce_rule->pattern.bMAC_SrcEnable = is_zero_ether_addr(key->src);
+		pce_rule->pattern.bMAC_DstEnable =
+						!is_zero_ether_addr(key->dst);
+		pce_rule->pattern.bMAC_SrcEnable =
+						!is_zero_ether_addr(key->src);
 
 		netdev_dbg(dev, "%s: bMAC_DstEnable = %d bMAC_SrcEnable = %d\n",
 			   __func__, pce_rule->pattern.bMAC_DstEnable,
@@ -303,10 +305,12 @@ static void pon_qos_trap_set_traffic_class(struct net_device *dev,
 					   GSW_PCE_rule_t *pce_rule,
 					   unsigned int classid)
 {
+	netdev_dbg(dev, "%s: eTrafficClassAction = 0x%x\n",
+		   __func__, classid);
 	if (classid != 0) {
 		pce_rule->action.eTrafficClassAction =
 			GSW_PCE_ACTION_TRAFFIC_CLASS_ALTERNATIVE;
-		pce_rule->action.nTrafficClassAlternate = classid;
+		pce_rule->action.nTrafficClassAlternate = classid & 0xf;
 
 		netdev_dbg(dev, "%s: eTrafficClassAction = %d\n",
 			   __func__, pce_rule->action.eTrafficClassAction);
